device/src/region/fixed_channel_plans/mod.rs

```diff
@@ -51,12 +70,15 @@ struct PreferredJoinChannels {
 impl PreferredJoinChannels {
     fn try_get_channel(&mut self, rng: &mut impl RngCore) -> Option<u8> {
         if self.num_retries > 0 {
-            let random = rng.next_u32();
+            let mut random = (rng.next_u32() & 0b111111) as usize;
```

Comment:
i carried over this method from elsewhere, but I do think there's room for a better method here. The key thing is "we store channel preferences in the ChannelMask struct consistently" and then separately, we can find better channel selection methods and we can apply that to all places that use ChannelMask for channel selection with rng.